apps/api/src/app/controllers/command/bulk-imports/arango-bulk-job-repository.ts

```diff
@@ -0,0 +1,205 @@
+import { Ack } from '@coscrad/commands';
+import { isNonEmptyString } from '@coscrad/validation-constraints';
+import { Inject } from '@nestjs/common';
+import { plainToClass } from 'class-transformer';
+import { AggregateId } from '../../../../domain/types/AggregateId';
+import { InternalError } from '../../../../lib/errors/InternalError';
+import { Maybe } from '../../../../lib/types/maybe';
+import { isNotFound, NotFound } from '../../../../lib/types/not-found';
+import cloneToPlainObject from '../../../../lib/utilities/cloneToPlainObject';
+import { ArangoConnectionProvider } from '../../../../persistence/database/arango-connection.provider';
+import { ArangoDatabase } from '../../../../persistence/database/arango-database';
+import { ArangoDatabaseForCollection } from '../../../../persistence/database/arango-database-for-collection';
+import mapDatabaseDocumentToAggregateDTO from '../../../../persistence/database/utilities/mapDatabaseDocumentToAggregateDTO';
+import mapEntityDTOToDatabaseDocument, {
+    ArangoDatabaseDocument,
+} from '../../../../persistence/database/utilities/mapEntityDTOToDatabaseDocument';
+import { DTO } from '../../../../types/DTO';
+import { ResultOrError } from '../../../../types/ResultOrError';
+import { CommandFSA } from '../command-fsa/command-fsa.entity';
+import { BulkCommandExecutionResult, CoscradBulkImportJob } from './bulk-import-job.entity';
+import { IBulkJobRepository } from './bulk-job-repository.interface';
+
+export const ARANGO_BULK_JOB_COLLECTION_NAME = 'bulk-import-jobs';
+
+export class ArangoBulkJobRepository implements IBulkJobRepository {
+    private readonly db: ArangoDatabaseForCollection<DTO<CoscradBulkImportJob>>;
+
+    constructor(@Inject(ArangoConnectionProvider) connectionProvider: ArangoConnectionProvider) {
+        const arangoDb = new ArangoDatabase(connectionProvider.getConnection());
+
+        this.db = new ArangoDatabaseForCollection(arangoDb, ARANGO_BULK_JOB_COLLECTION_NAME);
+    }
+
+    async create(bulkJob: CoscradBulkImportJob): Promise<AggregateId> {
+        const doc = this.instanceToArangoDoc(bulkJob);
+
+        const query = `
+            insert @newDoc in @@collectionName
```

Comment:
should we stick with capitalizing the aql statement keywords?

Reply:
I'd prefer to lower case them, if possible. It's really hard on the wrist. On a related note, I want to look into whether or not we can get intellisense similar to what we get for `tsx` in these snippets.